cli/process_utils/process_utils.go

```diff
@@ -0,0 +1,198 @@
+package process_utils
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"syscall"
+	"time"
+)
+
+const defaultDirPerms = 0770
+
+const (
+	ProcStateStopped = "NOT RUNNING."
+	ProcStateDead    = "ERROR. The process is dead."
+	ProcStateRunning = "RUNNING. PID: %v."
+)
+
+// GetPIDFromFile returns PID from the PIDFile.
+func GetPIDFromFile(pidFileName string) (int, error) {
+	if _, err := os.Stat(pidFileName); err != nil {
+		return 0, fmt.Errorf(`Can't "stat" the PID file. Error: "%v".`, err)
+	}
+
+	pidFile, err := os.Open(pidFileName)
+	if err != nil {
+		return 0, fmt.Errorf(`Can't open the PID file. Error: "%v".`, err)
+	}
+
+	pidBytes, err := ioutil.ReadAll(pidFile)
+	if err != nil {
+		return 0, fmt.Errorf(`Can't read the PID file. Error: "%v".`, err)
+	}
+
+	pid, err := strconv.Atoi(strings.TrimSpace(string(pidBytes)))
+	if err != nil {
+		return 0,
+			fmt.Errorf(`PID file exists with unknown format. Error: "%s"`, err)
+	}
+
+	return pid, nil
+}
+
+// CheckPIDFile checks that the process PID file exists
+// and is readable. Or process is already exist.
+// Removes PID file if process is dead.
+func CheckPIDFile(pidFileName string) error {
+	if _, err := os.Stat(pidFileName); err == nil {
+		// The PID file already exists. We have to check if the process is alive.
+		pid, err := GetPIDFromFile(pidFileName)
+		if err != nil {
+			return fmt.Errorf(`PID file exists, but PID can't be read. Error: "%v".`, err)
+		}
+		if res, _ := IsProcessAlive(pid); res {
+			return fmt.Errorf("The process is already exists. PID: %d", pid)
+		} else {
+			os.Remove(pidFileName)
+		}
+	} else if !os.IsNotExist(err) {
+		return fmt.Errorf(`Something went wrong while trying to read the PID file. Error: "%v".`,
+			err)
+	}
+
+	return nil
+}
+
+// CreatePIDFile checks that the instance PID file is absent or
+// deprecated and creates a new one. Returns an error on failure.
+func CreatePIDFile(pidFileName string) error {
+	if err := CheckPIDFile(pidFileName); err != nil {
+		return err
+	}
+
+	pidAbsDir := filepath.Dir(pidFileName)
+	if _, err := os.Stat(pidAbsDir); err != nil {
+		if os.IsNotExist(err) {
+			err = os.MkdirAll(pidAbsDir, defaultDirPerms)
+			if err != nil {
+				return fmt.Errorf(`can't crete PID file directory. Error: "%v".`, err)
+			}
+		} else {
+			return fmt.Errorf(`can't stat PID file directory. Error: "%v".`, err)
+		}
+	}
+
+	// Create a new PID file.
+	// 0644:
+	//    user:   read/write
+	//    group:  read
+	//    others: read
+	pidFile, err := os.OpenFile(pidFileName,
+		syscall.O_EXCL|syscall.O_CREAT|syscall.O_RDWR, 0644)
+	if err != nil {
+		return fmt.Errorf(`Can't create a new PID file. Error: "%v".`, err)
+	}
+	defer pidFile.Close()
+
+	if _, err = pidFile.WriteString(strconv.Itoa(os.Getpid())); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// StopProcess stops the process by pidFile.
+func StopProcess(pidFile, consoleSocket, processName string) error {
```

Comment:
Only the PID is needed to stop the process by PID, the other arguments must not be passed.

Reply:
Fixed `StopProcess`:
https://github.com/tarantool/tt/blob/f2224f765a8339ca1a8bedc9a8bdd3a32cea0284/cli/process_utils/process_utils.go#L113-L134